### PR TITLE
customer subscription details

### PR DESCRIPTION
**Pull request by ankitdas13**

### CODE REVIEW

File: Controller/Subscrib/View.php [hunk 1]
```diff
@@ -52,7 +52,6 @@ public function __construct(
      */
     public function execute()
     {
-        //echo "111";
```
Comment (HancilSequeira) [inline Controller/Subscrib/View.php:55]: Remove this

File: Helper/Subscription.php [hunk 2]
```diff
@@ -330,5 +330,52 @@ public function cancelSubscription($id, $rzp)
         $postUpdate->setCancelBy('customer');
 
         $postUpdate->save();
-    }        
+    }
+
+    public function pauseSubscription($id, $rzp)
```
Comment (HancilSequeira) [inline Helper/Subscription.php:335]: Add document comment explaining what this function does

File: Helper/Subscription.php [hunk 3]
```diff
@@ -330,5 +330,52 @@ public function cancelSubscription($id, $rzp)
...
+        $subscription = $this->_objectManagement->create('Razorpay\Subscription\Model\Subscriptions');
+        $postUpdate = $subscription->load($subscriptionResponse->id, 'subscription_id');
+
+        $postUpdate->setStatus('paused');
+        $postUpdate->setCancelBy('customer');
+
+        $postUpdate->save();
+    }
+
+    public function resumeSubscription($id, $rzp)
```
Comment (HancilSequeira) [inline Helper/Subscription.php:350]: Add document comment explaining what this function does

File: Helper/Subscription.php [hunk 4]
```diff
@@ -330,5 +330,52 @@ public function cancelSubscription($id, $rzp)
...
+        $subscription = $this->_objectManagement->create('Razorpay\Subscription\Model\Subscriptions');
+        $postUpdate = $subscription->load($subscriptionResponse->id, 'subscription_id');
+
+        $postUpdate->setStatus('active');
+        $postUpdate->setCancelBy('customer');
+
+        $postUpdate->save();
+    }
+
+    public function allSubscriptionInvoice($subscriptionId, $rzp){
```
Comment (HancilSequeira) [inline Helper/Subscription.php:365]: Change the function name to `fetchSubsciptionInvoice` and add a document comment explaining what this function does

**Review by HancilSequeira — CHANGES_REQUESTED** [review]

File: Controller/Subscrib/View.php (hunk 1 above)
Comment (ankitdas13) [reply Controller/Subscrib/View.php:55]: already removed

**Review by sofiya02 — APPROVED** [review]

File: Block/View.php [hunk 5]
```diff
@@ -104,8 +132,29 @@ protected function _getSubscribId()
...
         Subscrib $subscrib
     ) {
-        
+
         return '/razorpaysubscription/customer/CancelSubscription/s_id' . $subscrib->getSubscriptionId();
-        
+
+    }
+
+    public function _getCurrencySymbol($_code)
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply Block/View.php]: done

File: Helper/Subscription.php (hunk 4 above)
Comment (ankitdas13) [reply Helper/Subscription.php]: i have changed the function name `allSubscriptionInvoice` to `fetchSubsciptionInvoice `

File: view/frontend/templates/subscrib/view.phtml [hunk 6]
```diff
@@ -1,28 +1,30 @@
...
+
+          <?php if ($subscriptioninfo['status'] != 'paused'): ?>
+            <a href="<?php echo $block->getUrl('razorpaysubscription/customer/pauseSubscription/'); ?>s_id/<?php echo $subscriptioninfo['subscription_id']; ?>/oid/<?php echo $subscriptioninfo['entity_id']; ?>"
+               class="action primary"
+               style="float:right;"  >Pause</a>
+          <?php endif ?>
+
+            <a href="<?php echo $block->getUrl('razorpaysubscription/customer/resumeSubscription/'); ?>s_id/<?php echo $subscriptioninfo['subscription_id']; ?>/oid/<?php echo $subscriptioninfo['entity_id']; ?>"
+               class="action primary"
+               style="float:right;"  >Resume</a>
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply view/frontend/templates/subscrib/view.phtml:24]: i have added the condition

File: view/frontend/templates/subscrib/view.phtml [hunk 7]
```diff
@@ -31,72 +33,152 @@ $subscriptioninfo = $block->getSubscrib();
...
+                                </td>
+                                <td data-th="<?= $block->escapeHtmlAttr(__('Add-Ons')) ?>" class="col addon">--</td>
+                                <td data-th="<?= $block->escapeHtmlAttr(__('Status')) ?>" class="col status"><?= 'Next Due' ?></td>
+                                <td data-th="<?= $block->escapeHtmlAttr(__('Total amount')) ?>" class="col total"><?= $block->_getCurrencySymbol($subscriptioninfo['store_currency_code']).$invoices['items'][0]['amount_paid']/100 ?></td>
+                            </tr>
+                    <?php endif ?>
+                    <?php foreach ($invoices['items'] as $invoice) : ?>
+                            <tr>
+                                <td data-th="<?= $block->escapeHtmlAttr(__('Date')) ?>" class="col date"><?=  !empty($invoice['billing_start']) ? date('M d, Y', $invoice['billing_start']) : '--' ?></td>
+                                <td data-th="<?= $block->escapeHtmlAttr(__('Invoice Id')) ?>" class="col invoice"><a href="<?php $invoice['short_url'] ?>"><?php echo $invoice['id'] ?></a></td>
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply view/frontend/templates/subscrib/view.phtml]: url is fixed

File: view/frontend/templates/subscrib/view.phtml [hunk 8]
```diff
@@ -1,28 +1,30 @@
...
-                    Cancel </a> -->
-                    <a href="<?php echo $block->getUrl('razorpaysubscription/customer/cancelSubscription/'); ?>s_id/<?php echo $subscriptioninfo['subscription_id']; ?>/oid/<?php echo $subscriptioninfo['entity_id']; ?>" class="action primary"  style="float:right;" <?php if($subscriptioninfo['status']=='cancelled') echo 'disabled';?> >Cancel</a>
-     <a href="<?php echo $block->getUrl('razorpaysubscription/customer/cancelSubscription/'); ?>s_id/<?php echo $subscriptioninfo['subscription_id']; ?>/oid/<?php echo $subscriptioninfo['entity_id']; ?>" class="action primary" style="float:right;">Upgrade</a>
+    <div class="block-title"><strong>Order # <?php echo $subscriptioninfo['entity_id']; ?></strong>
+     <?= $subscriptioninfo['status'] ?>
+        <?php if ($subscriptioninfo['status'] != 'cancelled' && $subscriptioninfo['status'] != 'created'): ?>
+            <a href="<?php echo $block->getUrl('razorpaysubscription/customer/cancelSubscription/'); ?>s_id/<?php echo $subscriptioninfo['subscription_id']; ?>/oid/<?php echo $subscriptioninfo['entity_id']; ?>"
+               class="action primary"
+               style="float:right;"  >Cancel</a>
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply view/frontend/templates/subscrib/view.phtml]: i have changed the bg color of buttons (cancel & pause)

**Comment by sofiya02** [comment]
LGTM

**Review by sofiya02 — APPROVED** [review]

**Pull request merged**